### PR TITLE
add pkg-config/0.29.2

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **pkg-config/0.29.2**

Fixes https://github.com/conan-io/wishlist/issues/113

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/pkg-config/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    settings = "os_build", "arch_build", "compiler"
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os_build == "Windows" and self.settings.compiler == "gcc" and os.name == "nt"
+
+    def build_requirements(self):
+        if self._is_mingw_windows and "CONAN_BASH_PATH" not in os.environ:
```
Comment (danimtb) [inline recipes/pkg-config/all/conanfile.py]: This will need an additional check for a system env var

File: recipes/pkg-config/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.freedesktop.org/wiki/Software/pkg-config/"
+    license = "GPL-2.0-or-later"
+    settings = "os_build", "arch_build", "compiler"
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os_build == "Windows" and self.settings.compiler == "gcc" and os.name == "nt"
```
Comment (danimtb) [inline recipes/pkg-config/all/conanfile.py]: not sure about the os.name check

Comment (ericLemanissier) [reply recipes/pkg-config/all/conanfile.py]: `os.name == 'posix'` would be cygwin environment

File: recipes/pkg-config/all/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/pkg-config/all/conanfile.py]: I don't understand what env var you are talking about. I can add ` and os_info.detect_windows_subsystem() != 'msys2'`

Comment (danimtb) [reply recipes/pkg-config/all/conanfile.py]: yes, that was it

**Comment by SSE4** [comment]
for me the main concern is missing Windows build, which is most important.
I'd say we have to use method described here:
https://cgit.freedesktop.org/pkg-config/tree/README.win32
https://cgit.freedesktop.org/pkg-config/tree/Makefile.vc

File: recipes/pkg-config/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("pkg-config-" + self.version, self._source_subfolder)
+
+    def build(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("builds with Visual Studio aren't currently supported, "
```
Comment (danimtb) [inline recipes/pkg-config/all/conanfile.py]: I missed this! Should be in configure() step IMO

**Comment by ericLemanissier** [comment]
So, building pkg-config with msvc requires an existing build of glib (https://cgit.freedesktop.org/pkg-config/tree/README.win32#n30 and https://cgit.freedesktop.org/pkg-config/tree/Makefile.vc#n1). Glib has several requirements, which are not yet on CCI (libselinux, gettext, flex and automake_build_aux). I'll try to do transfer these recipes before pkg-config

**Comment by SSE4** [comment]
@ericLemanissier moreover, this going to be a circular dependency: **Glib** -> **pkg-config** -> **Glib**... I am not sure how can we deal with that

**Comment by ericLemanissier** [comment]
According to pkg-config doc: `Note that MSVC
builds of glib does not have a build-time dependency on pkg-config`

**Pull request closed without merging**